### PR TITLE
Autcomplete feature

### PR DESCRIPTION
**Pull request by joshlopes**
Q   |   A
------|-------
Bug fix? |	no
New feature? |	yes
BC breaks? | ?
Deprecations? | ? 
Tests pass? | ?
Fixed tickets | 
License | yes
Doc PR | yes

### CODE REVIEW

**Comment by joshlopes** [comment]
This comes from the old PR #137 , please review & if possible fix tests in the new merge thingy or i will fix it later when i get myself some time.

**Comment by Soullivaneuh** [comment]
> this SonataCI - Quick Check is new ?

Yes, new tools under testing. Check for Travis for now.

CS need to be fixe, please rebase your PR to make travis green.

File: Builder/DatagridBuilder.php [hunk 1]
```diff
@@ -116,6 +121,16 @@ public function addFilter(DatagridInterface $datagrid, $type = null, FieldDescri
         $admin->addFilterFieldDescription($fieldDescription->getName(), $fieldDescription);
 
         $fieldDescription->mergeOption('field_options', array('required' => false));
+
+        if ($type === 'doctrine_mongo_autocomplete') {
```
Comment (Soullivaneuh) [inline Builder/DatagridBuilder.php]: Yoda condition.

File: Builder/DatagridBuilder.php [hunk 2]
```diff
@@ -145,4 +160,23 @@ public function getBaseDatagrid(AdminInterface $admin, array $values = array())
 
         return new Datagrid($admin->createQuery(), $admin->getList(), $pager, $formBuilder, $values);
     }
+
+    /**
+     * Get pager by pagerType
+     *
+     * @param string $pagerType
+     *
+     * @return \Sonata\AdminBundle\Datagrid\PagerInterface
```
Comment (Soullivaneuh) [inline Builder/DatagridBuilder.php]: Do not use FQN, import instead

File: Filter/Filter.php [hunk 3]
```diff
@@ -28,6 +28,27 @@ public function apply($queryBuilder, $value)
         $this->filter($queryBuilder, null, $this->getFieldName(), $value);
     }
```
Comment (Soullivaneuh) [inline Filter/Filter.php]: Missing doc block.

File: Filter/ModelAutocompleteFilter.php [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DoctrineMongoDBAdminBundle\Filter;
+
+use Sonata\CoreBundle\Form\Type\EqualType;
+use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
+use Doctrine\Common\Collections\Collection;
+
```
Comment (Soullivaneuh) [inline Filter/ModelAutocompleteFilter.php]: Missing class comment.

File: Filter/ModelAutocompleteFilter.php [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+class ModelAutocompleteFilter extends Filter
+{
+
+    public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
```
Comment (Soullivaneuh) [inline Filter/ModelAutocompleteFilter.php]: Missing doc block.

File: Filter/ModelAutocompleteFilter.php [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+        if (is_array($data['value'])) {
+            $this->handleMultiple($queryBuilder, $alias, $field, $data);
+        } else {
+            $this->handleScalar($queryBuilder, $alias, $field, $data);
+        }
+
+    }
+
+    /**
+     *
```
Comment (Soullivaneuh) [inline Filter/ModelAutocompleteFilter.php]: Blank line not needed.

File: Filter/ModelAutocompleteFilter.php [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+     * @param  ProxyQueryInterface $queryBuilder
+     * @param  type                $alias
+     * @param  type                $field
+     * @param  type                $data
+     * @return type
```
Comment (Soullivaneuh) [inline Filter/ModelAutocompleteFilter.php]: Blank line needed before return.

File: Filter/ModelAutocompleteFilter.php [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+        $this->active = true;
+    }
+
+    /**
+     *
+     * @param  ProxyQueryInterface $queryBuilder
+     * @param  type                $alias
+     * @param  type                $field
+     * @param  type                $data
+     * @return type
```
Comment (Soullivaneuh) [inline Filter/ModelAutocompleteFilter.php]: Blank line needed before return.

File: composer.json [hunk 9]
```diff
@@ -23,6 +23,7 @@
     },
     "require-dev": {
         "jmikola/geojson": "~1.0",
+        "phpunit/phpunit": "~4.0",
```
Comment (Soullivaneuh) [inline composer.json]: Please remove this, not needed.

**Comment by joshlopes** [comment]
@Soullivaneuh done.

File: Filter/Filter.php [hunk 10]
```diff
@@ -19,17 +19,35 @@
     protected $active = false;
 
     /**
-     * {@inheritdoc}
+     * @param mixed $queryBuilder
```
Comment (Soullivaneuh) [inline Filter/Filter.php]: Why removing `inheritdoc`?

Comment (joshlopes) [inline Filter/Filter.php]: No reason, should i put it back ?

Comment (Soullivaneuh) [inline Filter/Filter.php]: Of course. If this method as a parent, it must have this tag.

**Comment by Soullivaneuh** [comment]
According to the new [Sonata version management](https://github.com/sonata-project/SonataAdminBundle/issues/3053) and next major [release plan](https://github.com/sonata-project/SonataAdminBundle/issues/3731), this project has been refactored regarding branching and versioning.

If you see this message, your PR concerns a patch or a minor release and is not targeting the right branch.

So I'm closing this one, but don't see it as a refusal. If you think your work is still relevant and want to continue, feel free to reopen it on the right branch (e.g. the default one).

Regards.

**Pull request closed without merging**